src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const {
+  getUserById,
+} = require('../services/users.service');
+
+const {
+  getAllExpenses,
+  getExpenseById,
+  addNewExpense,
+  removeExpense,
+  updateExpense,
+} = require('../services/expenses.service');
+
+const get = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const expenses = getAllExpenses();
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses
+      = filteredExpenses.filter((expense) => expense.userId === +userId);
+
+    if (!filteredExpenses.length) {
+      res.sendStatus(404);
+
+      return;
+    }
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter((expense) =>
+      categories.includes(expense.category)
+    );
+
+    if (!filteredExpenses.length) {
+      res.sendStatus(404);
+
+      return;
+    }
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) =>
+        new Date(expense.spentAt) >= new Date(from)
+        && new Date(expense.spentAt) <= new Date(to)
+    );
+
+    if (!filteredExpenses.length) {
+      res.sendStatus(404);
+
+      return;
+    }
+  }
+
+  res.send(filteredExpenses);
+};
+
+const add = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const user = getUserById(+userId);
+
+  if (!user
+    || !spentAt
+    || !title
+    || !amount
+    || !category
+    || !note
+  ) {
+    res.sendStatus(400);
```

Comment:
If you don't have some of the required fields you don't need to get a user from the database.
So it should look like: validate inputs => if ok get user => if user in the database create expense